derive_builder_core/src/macro_options/darling_opts.rs

```diff
@@ -92,6 +92,11 @@ impl FlagVisibility for BuildFn {
 pub struct FieldMeta {
     public: Flag,
     private: Flag,
+    /// Custom builder field type
+    #[darling(rename = "type")]
+    builder_type: Option<syn::Type>,
+    /// Custom builder field method, for making target struct field value
+    build: Option<BlockContents>,
```

Comment:
I had forgotten that `FieldMeta` is currently used by both `Field` and `Options` - these settings definitely don't make sense in the latter context, so we'll need to follow the example of `StructLevelSetter` and `FieldLevelSetter` and make two structs. The struct-level one should only have the visibility settings, while the field level one would have the visibility settings and these new fields.

```rust
/// Contents of the `field` meta in `builder` attributes.
#[derive(Debug, Clone, Default, FromMeta)]
pub struct StructLevelFieldMeta {
    public: Flag,
    private: Flag,
    vis: Option<syn::Visibility>,
}

impl Visibility for StructLevelFieldMeta {
    fn public(&self) -> &Flag {
        &self.public
    }

    fn private(&self) -> &Flag {
        &self.private
    }

    fn explicit(&self) -> Option<&syn::Visibility> {
        self.vis.as_ref()
    }
}

#[derive(Debug, Clone, Default, FromMeta)]
pub struct StructLevelFieldMeta {
    public: Flag,
    private: Flag,
    vis: Option<syn::Visibility>,
    /// Custom builder field type
    #[darling(rename = "type")]
    builder_type: Option<syn::Type>,
    /// Custom builder field method, for making target struct field value
    build: Option<BlockContents>,
}

impl Visibility for FieldLevelFieldMeta {
    fn public(&self) -> &Flag {
        &self.public
    }

    fn private(&self) -> &Flag {
        &self.private
    }

    fn explicit(&self) -> Option<&syn::Visibility> {
        self.vis.as_ref()
    }
}
```

Reply:
Oh.  I think the struct-level ones will always be a subset of the field-level ones, with the same names, surely?  Anything else would be very confusing.  Unfortunately darling doesn't have `#[darling(flatten)]`.

Reply:
Yes, but reusing the same struct would cause darling’s “did you mean” suggestions to include fields that aren’t valid at the struct level. The code duplication is annoying, but at least it’s fairly small.

Reply:
I think at this point I need to rebase onto #247, before doing as you suggest.  Is #247 likely to change significantly, or can I do that now?

Reply:
I've just merged it (and the last push was just a commit message edit, so it hasn't changed at all).

Reply:
Or, maybe, merge with #247.